src/Network/XMPP/MD5.hs

```diff
@@ -101,29 +103,29 @@ instance MD5 WordList where
   where (xs, ys) = splitAt 16 ws
         taken = if l > 511 then 512 else l .&. 511
  len_pad c64 (WordList (ws, l)) = WordList (beginning ++ nextish ++ blanks ++ size, newlen)
-  where beginning = if length ws > 0 then start ++ lastone' else []
+  where beginning = if not $ null ws then start ++ lastone' else []
         start = init ws
         lastone = last ws
         offset = c64 .&. 31
         lastone' = [if offset > 0 then lastone + theone else lastone]
         theone = shiftL (shiftR 128 (fromIntegral $ offset .&. 7))
                         (fromIntegral $ offset .&. (31 - 7))
-        nextish = if offset == 0 then [128] else []
+        nextish = [128 | offset == 0]
         c64' = c64 + (32 - offset)
-        num_blanks = (fromIntegral $ shiftR ((448 - c64') .&. 511) 5)
+        num_blanks = fromIntegral $ shiftR ((448 - c64') .&. 511) 5
         blanks = replicate num_blanks 0
         lowsize = fromIntegral $ c64 .&. (shiftL 1 32 - 1)
         topsize = fromIntegral $ shiftR c64 32
         size = [lowsize, topsize]
-        newlen = l .&. (complement 511)
+        newlen = l .&. complement 511
                + if c64 .&. 511 >= 448 then 1024 else 512
  finished (WordList (_, z)) = z == 0
 
 
 instance Num ABCD where
```

Comment:
This whole instance is massively fishy, we should either find nice algebraic laws for it or get rid of it entirely. Promising a Num, but being undefined on 3/4 of it is just... crazy...